source/mqtt/Mqtt5Client.cpp

```diff
@@ -171,46 +173,45 @@ namespace Aws
-                    switch (packet_type)
+                    case aws_mqtt5_packet_type::AWS_MQTT5_PT_PUBACK:
                     {
-                        case aws_mqtt5_packet_type::AWS_MQTT5_PT_PUBACK:
+                        if (publshCompletionPacket != NULL)
                         {
-                            if (publshCompletionPacket != NULL)
-                            {
-                                std::shared_ptr<PubAckPacket> packet = std::make_shared<PubAckPacket>(
-                                    *(aws_mqtt5_packet_puback_view *)publshCompletionPacket, callbackData->allocator);
-                                publish = std::make_shared<PublishResult>(std::move(packet));
-                            }
-                            else // This should never happened.
-                            {
-                                AWS_LOGF_INFO(AWS_LS_MQTT5_CLIENT, "The PubAck Packet is invalid.");
-                                publish = std::make_shared<PublishResult>(AWS_ERROR_INVALID_ARGUMENT);
-                            }
-                            break;
-                        }
-                        case aws_mqtt5_packet_type::AWS_MQTT5_PT_NONE:
-                        {
-                            publish = std::make_shared<PublishResult>(error_code);
-                            break;
+                            std::shared_ptr<PubAckPacket> packet = std::make_shared<PubAckPacket>(
+                                *(aws_mqtt5_packet_puback_view *)publshCompletionPacket, callbackData->allocator);
+                            publish = std::make_shared<PublishResult>(std::move(packet));
                         }
-                        default: // Invalid packet type
+                        else /* This should never happened. */
                         {
-                            AWS_LOGF_INFO(AWS_LS_MQTT5_CLIENT, "Invalid Packet Type.");
+                            AWS_LOGF_INFO(AWS_LS_MQTT5_CLIENT, "The PubAck Packet is invalid.");
                             publish = std::make_shared<PublishResult>(AWS_ERROR_INVALID_ARGUMENT);
-                            break;
                         }
+                        break;
                     }
-                    if (callbackData->onPublishCompletion != NULL)
+                    case aws_mqtt5_packet_type::AWS_MQTT5_PT_NONE:
                     {
-                        callbackData->onPublishCompletion(callbackData->client, error_code, publish);
+                        publish = std::make_shared<PublishResult>(error_code);
+                        break;
                     }
-
-                    Crt::Delete(callbackData, callbackData->allocator);
+                    default: /* Invalid packet type */
+                    {
+                        AWS_LOGF_INFO(AWS_LS_MQTT5_CLIENT, "Invalid Packet Type.");
+                        publish = std::make_shared<PublishResult>(AWS_ERROR_INVALID_ARGUMENT);
```

Comment:
Invalid argument is intended for user-submitted data, not crt-generated data.  I would use a different error code here and above.

Reply:
I checked out the error codes, but I did not find anything suitable for the case here, I used AWS_ERROR_UNKNOWN for now.